File: abc258/d.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll N, X;
    cin >> N >> X;
    vector<pair<ll, ll>> stages;
    for (int i = 0; i < N; i++)
    {
        ll a, b;
        cin >> a >> b;
        stages.push_back(pair{a, b});
    }
    ll cost_base = 0;
    ll B_min = 1e10;
    ll ans = 5e18;
    for (int i = 0; i < N; i++)
    {
        auto stage = stages[i];
        ll A = stage.first;
        ll B = stage.second;
        cost_base += (A + B);
        B_min = min(B_min, B);
        ll total_cost = cost_base + (B_min * (X - (i + 1)));
        ans = min(ans, total_cost);
    }

    cout << ans << endl;
}

File: abc258/d.py
N,X = map(int,input().split())

# cost_base = 0
A = []
B = []
for _ in range(N):
    a,b = map(int,input().split())
    A.append(a)
    B.append(b)

cost_base = 0
B_min = 5e18
ans = 5e18
for i in range(N):
    a,b = A[i],B[i]
    cost_base += a + b
    B_min = min(B_min,b)
    stage_base = i + 1
    # 残りのステージをB_minでやるのが最良
    if stage_base > X:
        break
    ans = min(ans,B_min * (X - stage_base) + cost_base)

print(ans)
